packages/ui/src/components/tooltip/tooltip.component.tsx

```diff
@@ -0,0 +1,55 @@
+import { useEffect, useId, useState } from 'react';
+import { mergeProps, useHover, useFocusRing } from 'react-aria';
+
+import { TooltipContent } from './components/tooltip-content/tooltip-content.component.js';
+import { styles } from './tooltip.styles.js';
+import { TooltipProps } from './tooltip.types.js';
+
+// TODO: Complete component/replace with a library that works (react-tooltip if can find a way to remove aria-described by for select)
+export function Tooltip({ children, tooltip, id, className }: TooltipProps) {
+  const localId = useId();
+  const [isOpen, setIsOpen] = useState(false);
+  let tooltipWaitTime: NodeJS.Timeout;
+
+  const startTimer = () => {
+    tooltipWaitTime = setTimeout(() => {
```

Comment:
Declaring `let tooltipWaitTime` inside the component body means it’s redefined on every render.

So when stopTimer runs, it might clear a different timeout reference.
can you use `useRef` for the timeout.